src/stateToHTML.js

```diff
@@ -217,7 +217,10 @@ class MarkupGenerator {
       this.currentBlock += 1;
       return;
     }
-    this.writeStartTag(blockType, customRenderer, block.data);
+
+    if (block.text.trim() !== '' || isEmptyBlock(blockType)) {
```

Comment:
If the text of the block is not empty, or it is an empty block type (such as <hr>), then the start tag should be rendered.